### DISCUSSION TITLE
Text Color Not Inverting in Dark and Light Theme

### DISCUSSION THREAD

**Discussion by sarsw** [MahApps/MahApps.Metro#4098] [Q&A] [answered]
Hi,

My WPF app needs to support theming, I've also just migrated from 1.6 to 2.4.5.  I can change themes and accents and my elements and backgrounds follow light & dark as expected...mostly, flyouts have transparent elements since migrating!  My text however is not inverting to remain visible. My users can configure the color of text themselves or use a theme color so I'm reading the text as a string and applying it, e.g.

tb.Foreground = new BrushConverter().ConvertFromString(textcolor) as SolidColorBrush;   // for regular colors like red, blue or hex
or
tb.Foreground = (Brush)FindResource(textColor);   // for a mahapps theme color

In pre 2.0 I [read](https://github.com/MahApps/MahApps.Metro/issues/3541) that BlackBrush should be used, in post 2.0 this brush is called MahApps.Brushes.ThemeForeground (found [here](https://mahapps.com/docs/guides/migration-to-v2.0)).

For me, the text just stays black and does not invert.

I've tried a few other MahApps.Brushes too and none invert to be readable in both dark and light uses.  What should I be using?

I'm using Mahapps.Metro 2.4.5, W10x64, VS2017, .Net 4.6.1.

Regards,
Steve

**Comment by timunie** [comment, marked as answer]
Hi @sarsw 

try this: 

```c#
private void ColorPicker_SelectedColorChanged(object sender, System.Windows.RoutedPropertyChangedEventArgs<System.Windows.Media.Color?> e)
{
    App.Current.Resources["MahApps.Brushes.Text"] = new SolidColorBrush(e.NewValue ?? Colors.Black);
    App.Current.Resources["MahApps.Brushes.ThemeForeground"] = new SolidColorBrush(e.NewValue ?? Colors.Black);
}
```

result: 
![image](https://user-images.githubusercontent.com/47110241/116744523-afb93200-a9fa-11eb-853d-56dd6afca6c7.png)

I found the needed keys here: 
https://github.com/MahApps/MahApps.Metro/blob/d25cbe6e288d91268f7e46e0ccf998ac0ade53c3/src/MahApps.Metro/Styles/Themes/Theme.Template.xaml#L67-L69

Happy coding
Tim

**Comment by sarsw** [comment]
Thanks, "App.Current.Resources["MahApps.Brushes.Text"] = new SolidColorBrush(e.NewValue ?? Colors.Black);" set me down the right path.   At startup I detect the Dark or Light mode in use (theme.BaseColorScheme) and set my theme brush resource accordingly, that gets picked up when used in the displays,

👍 

Steve
